backend/common.go

```diff
@@ -232,7 +233,7 @@ func propagateTenantIDIfPresent(ctx context.Context) context.Context {
 	return ctx
 }
 
-func (s *DataSourceInstanceSettings) ProxyOptions() (*proxy.Options, error) {
+func (s *DataSourceInstanceSettings) ProxyOptions(clientCfg *proxy.ClientCfg) (*proxy.Options, error) {
```

Comment:
this and the other breaking changes in the other hand are fine IMO since they are not that much used